kratos/utilities/read_materials_utility.cpp

```diff
@@ -156,45 +135,58 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
             const Variable<int>& variable = KratosComponents<Variable<int>>().Get(variable_name);
-            p_prop->SetValue(variable, value.GetInt());
+            if (!value.IsNull())
+                pNewProperty->SetValue(variable, value.GetInt());
         } else if(KratosComponents<Variable<array_1d<double, 3> > >::Has(variable_name)) {
             const Variable<array_1d<double, 3>>& variable = KratosComponents<Variable<array_1d<double, 3>>>().Get(variable_name);
-            array_1d<double, 3> temp(3, 0.0);
-            const Vector& value_variable = value.GetVector();
-            KRATOS_ERROR_IF(value_variable.size() != 3) << "The vector of variable " << variable_name << " has size " << value_variable.size() << " and it is supposed to be 3" << std::endl;
-            for (IndexType index = 0; index < 3; ++index)
-                temp[index] = value_variable[index];
-            p_prop->SetValue(variable, temp);
+            if (!value.IsNull()) {
+                array_1d<double, 3> temp(3, 0.0);
+                const Vector& value_variable = value.GetVector();
+                KRATOS_ERROR_IF(value_variable.size() != 3) << "The vector of variable " << variable_name << " has size " << value_variable.size() << " and it is supposed to be 3" << std::endl;
+                for (IndexType index = 0; index < 3; ++index)
+                    temp[index] = value_variable[index];
+                pNewProperty->SetValue(variable, temp);
+            }
         } else if(KratosComponents<Variable<array_1d<double, 6> > >::Has(variable_name)) {
             const Variable<array_1d<double, 6>>& variable = KratosComponents<Variable<array_1d<double, 6>>>().Get(variable_name);
-            array_1d<double, 6> temp(6, 0.0);
-            const Vector& value_variable = value.GetVector();
-            KRATOS_ERROR_IF(value_variable.size() != 6) << "The vector of variable " << variable_name << " has size " << value_variable.size() << " and it is supposed to be 6" << std::endl;
-            for (IndexType index = 0; index < 6; ++index)
-                temp[index] = value_variable[index];
-            p_prop->SetValue(variable, temp);
+            if (!value.IsNull()) {
+                array_1d<double, 6> temp(6, 0.0);
+                const Vector& value_variable = value.GetVector();
+                KRATOS_ERROR_IF(value_variable.size() != 6) << "The vector of variable " << variable_name << " has size " << value_variable.size() << " and it is supposed to be 6" << std::endl;
+                for (IndexType index = 0; index < 6; ++index)
+                    temp[index] = value_variable[index];
+                pNewProperty->SetValue(variable, temp);
+            }
         } else if(KratosComponents<Variable<Vector > >::Has(variable_name)) {
             const Variable<Vector>& variable = KratosComponents<Variable<Vector>>().Get(variable_name);
-            p_prop->SetValue(variable, value.GetVector());
+            if (!value.IsNull()) {
```

Comment:
why do you check for is null? it should be "IsVector" no?

Reply:
I don't remember the reason, but there was corner case

Reply:
For the proposition of @frastellini of assign null values (which means to take the computed value between properties)

Reply:
I am not sure if assigning null-values is the way to go for this
1. Your approach is not assigning anything if it is `null`, hence it is doing nothing => i.e. not doing what you intend
2. why cannot you explicitly compute the values that are to be computed from the subprops? I guess you have to do this, but why assigning null?
3. if we want to use null for sth else then your mechanism is broken.

Reply:
My proposal was setting a null to a property variable to indicate that this variable will be available as a variable of the property, but that its value is not know before hand since it will be calculated in terms of the sub-properties values.
We may discuss the better way of doing this, if the null definition is not well accepted by rest of developers. Any best proposal?

Reply:
I understand, but I don't like having unused settings in this file. Reason is that for more complicated models this file can also become very large and then it is easy to loose the overview. If one then also has to know if there are some specific settings that will be ignored (=> `null`) then it gets even more confusing.
Plus what if in the future `null` will get a meaning? Then your mechanism is broken.

Unfortunately I also don't have a better suggestion. :/

If you want to know which variables will be available later then I recommend that you at least check if a variable, that will be computed by a subprop, has a value assigned and throw if it is not null.
Otherwise what value would this variable get? The one the user specified or the one computed by the subprops?

Reply:
The user cannot specify its value since it will depend by the sub properties.
Maybe a better option is to set a 0 (zero) to avoid the null setting. In this case, zero will indicate the variable exists, but depends on the subproperties... 
what do you think? (@loumalouomega @RiccardoRossi @maceligueta @philbucher )

Reply:
hm I think zero is not better bcs this is actually a valid value.

Idea: Maybe here it would be a good place to use some enhanced syntax => #653

lets say `THICKNESS` is coming from a subproperty then you should do the following to indicate that it will be computed by a subproperty:
~~~json
THICKNESS : "@subprop"
~~~
the name is just a placeholder but gives an idea
this would also be easier to check in the readmaterialsprocess

what do you think?

Reply:
At the end the concept is the same, we can discuss it, but null is already a Parameter allowed value

Reply:
Maybe wait until #58 is merged (or #3341)

Reply:
My first initial idea was to use a string like `"auto"` to indicate this. Later @loumalouomega suggested me the use of `null` instead.
I agree with @philbucher proposal. Maybe we can consider alternative keywords like `"@automatic"` or `"@auto"` that could be employed to indicate the those variables will exist but its value will be calculated automatically by the CL in terms of the other variables (or subproperties' variables)

Reply:
I suggested null because was already on the interface, I can agree with the @auto , but I think that this is for a future PR

Reply:
ping to get opinion from @AlejandroCornejo who is also working with composite CLs

Reply:
i don't really understand when you would set a variable to null. In particular as if it is null you are simply doing nothing.
What would be the "@automatic" behaviour? who should set the defaults?

from a coding point of view in any case instead of checking for IsNull within each of the if clauses i would put it at the beginning of the if list. smthg like

     if( ! value.IsNull() )
     {
           if array ...
           else if Matrix ...
           else if double ...
           ...
     }
     else //(the variable was null)
     {
          //what do i do here??
     }

Reply:
@frastellini i think you are mixing up things here. A constitutive law cannot change properties...it can only change its internal variables!!

Reply:
Ok. I understand the design limitation.